src/bin/load_test.rs

```diff
@@ -199,39 +236,170 @@ async fn main() {
 
     let document_type = data_contract
         .document_type_cloned_for_name("preorder")
-        .unwrap();
+        .unwrap()
+        .into();
 
     let identity_lock = backend.state().loaded_identity.lock().await;
     let identity = identity_lock.as_ref().expect("no loaded identity");
 
     let identity_private_keys_lock = backend.state().identity_private_keys.lock().await;
 
+    let mut signer = SimpleSigner::default();
+
+    for (key_id, identity_public_key) in identity.public_keys() {
+        let private_key = identity_private_keys_lock
+            .get(&(identity.id(), *key_id))
+            .expect("expected a private key")
+            .clone();
+        signer.add_key(identity_public_key.clone(), private_key);
+    }
+
+    let arc_signer = Arc::new(signer);
+
+    // to be safe we need at least one contract per second of broadcasting,
+    // so if we are aiming at 1000 tx/s we would need 1000 contracts
+
+    let contract_count = args.contracts;
+
+    let document_types = broadcast_contract_variants(
+        Arc::clone(&sdk),
+        &identity,
+        arc_signer.clone(),
+        &data_contract,
+        contract_count,
+        9875, //let's use the same seed so we don't need to reregister contracts
+    )
+    .await
+    .into_iter()
+    .map(|contract| {
+        contract
+            .document_type_cloned_for_name("preorder")
+            .expect("expected preorder document")
+            .into()
+    })
+    .collect::<Vec<Arc<_>>>();
+
     broadcast_random_documents_load_test(
         Arc::clone(&sdk),
         &identity,
-        &identity_private_keys_lock,
-        Arc::new(data_contract),
-        Arc::new(document_type),
+        arc_signer,
+        document_type,
+        document_types,
         Duration::from_secs(args.time.into()),
         args.connections,
         args.rate,
     )
     .await;
 }
 
+async fn broadcast_contract_variants(
```

Comment:
Please add some logging to this function so that we know what we are doing, and if it hangs we can investigate where and why.